programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -160,7 +164,12 @@ impl<'a, 'b> CallerAccount<'a, 'b> {
                 account_info.data.as_ptr() as *const _ as u64,
                 invoke_context.get_check_aligned(),
             )?;
+
             if direct_mapping {
+                if account_info.data.as_ptr() as u64 >= ebpf::MM_INPUT_START {
+                    return Err(SyscallError::InvalidPointer.into());
+                }
```

Comment:
This check should be done before `data` pointer is dereferenced.

Reply:
Belt and braces, but yes you are right. Fixed